src/Microsoft.Graph.Core/Requests/Content/BatchResponseContent.cs

```diff
@@ -86,17 +82,18 @@ public async Task<HttpResponseMessage> GetResponseByIdAsync(string requestId)
         /// Gets a batch response as a requested type for the specified batch request id.
         /// </summary>
         /// <param name="requestId">A batch request id.</param>
-        /// <param name="responseHandler">ResponseHandler to use for the resonse</param>
+        /// <param name="errorMappings">A batch request id.</param>
+        /// <param name="responseHandler">ResponseHandler to use for the response</param>
         /// <returns>A deserialized object of type T<see cref="HttpResponseMessage"/>.</returns>
-        public async Task<T> GetResponseByIdAsync<T>(string requestId, IResponseHandler responseHandler = null) where T : IParsable, new()
+        public async Task<T> GetResponseByIdAsync<T>(string requestId, Dictionary<string, ParsableFactory<IParsable>> errorMappings = null, IResponseHandler responseHandler = null) where T : IParsable, new()
```

Comment:
so here we're expecting the caller to pass that mapping to us if I'm not mistaking? 
What about pushing that argument to the constructor, storing the value as a private field.
And then have derived classes in the service lib instead?

Reply:
https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/c065dd3405174a6aad94e85859b7abb1b673baa5/src/Microsoft.Graph/GraphServiceClient.cs#L92

Reply:
Thanks for clearing this up. 
The latest update reverts this and makes it a constructor variable. https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/607/commits/837b328b02764447041978953f3d94f426c92272
The service library can override this in its request builder here https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/c065dd3405174a6aad94e85859b7abb1b673baa5/src/Microsoft.Graph/GraphServiceClient.cs#L92